tests/test_plugin_loader.py

```
@@ -4815,6 +5072,128 @@ def test_update_existing_repo_to_branch_or_tag_fetch_failure(
+    def test_resolve_remote_default_branch_parses_symref_output(self, mock_run_git):
+        """Default branch resolver should parse `git ls-remote --symref` output."""
+        mock_run_git.return_value = subprocess.CompletedProcess(
+            args=[],
+            returncode=0,
+            stdout=(
+                "ref: refs/heads/main\tHEAD\n"
+                "0123456789abcdef0123456789abcdef01234567\tHEAD\n"
+            ),
+            stderr="",
+        )
+        self.assertEqual(pl._resolve_remote_default_branch("/tmp/repo"), "main")
```

Ruff (0.15.10)

[error] Probable insecure usage of temporary file or directory: "/tmp/repo"

(S108)